使用MinGW编译FLTK报错：narrowing conversion

提交 5dc031c 触发的Actions流水线报错：

<https://github.com/ZZy979/PPP-code/actions/runs/6291461674/job/17079850502>

```
In file included from C:/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from D:\a\PPP-code\PPP-code\cmake-build\_deps\fltk-src\fluid\ExternalCodeEditor_WIN32.h:27,
                 from D:\a\PPP-code\PPP-code\cmake-build\_deps\fltk-src\fluid\ExternalCodeEditor_WIN32.cxx:14:
D:\a\PPP-code\PPP-code\cmake-build\_deps\fltk-src\fluid\ExternalCodeEditor_WIN32.cxx: In member function 'int ExternalCodeEditor::reap_editor(DWORD*)':
D:\a\PPP-code\PPP-code\cmake-build\_deps\fltk-src\fluid\ExternalCodeEditor_WIN32.cxx:491:10: error: narrowing conversion of '4294967295' from 'DWORD' {aka 'long unsigned int'} to 'int' [-Wnarrowing]
  491 |     case WAIT_FAILED: {    // failed
      |          ^~~~~~~~~~~
mingw32-make[2]: *** [_deps\fltk-build\fluid\CMakeFiles\fluid.dir\build.make:361: _deps/fltk-build/fluid/CMakeFiles/fluid.dir/ExternalCodeEditor_WIN32.cxx.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:1451: _deps/fltk-build/fluid/CMakeFiles/fluid.dir/all] Error 2
```

从错误信息判断原因是FLTK源代码fluid\ExternalCodeEditor_WIN32.cxx中的缩小转换导致的。但是此前一直正常，本次提交运行的其他job也没报错，只有MinGW构建失败。

FLTK仓库已有相关issue：fltk/fltk#92、fltk/fltk#178、fltk/fltk#445，该bug已在1.4版本中修复，但目前尚未发布。

本地复现：

```cpp
int main() {
    const unsigned long a = 0xFFFFFFFF;
    int b = -1;
    switch (b) {
        case a:
            cout << 1 << endl;
            break;
        default:
            cout << 0 << endl;
            break;
    }
    return 0;
}
```

报错信息：

```
D:\CLionProjects\Cpp\main.cpp: In function 'int main()':
D:\CLionProjects\Cpp\main.cpp:9:14: error: narrowing conversion of '4294967295' from 'long unsigned int' to 'int' [-Wnarrowing]
    9 |         case a:
      |              ^
mingw32-make[3]: *** [CMakeFiles\main.dir\build.make:75: CMakeFiles/main.dir/main.cpp.obj] Error 1
mingw32-make[2]: *** [CMakeFiles\Makefile2:82: CMakeFiles/main.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:89: CMakeFiles/main.dir/rule] Error 2
mingw32-make: *** [Makefile:123: main] Error 2
```

但是本地使用MinGW编译FLTK并未复现该问题。

--- a/gui.cmake
+++ b/gui.cmake
@@ -16,6 +16,11 @@ else()
   set(GUI_TYPE)
 endif()
 
+# 绕过FLTK的bug：https://github.com/ZZy979/PPP-code/issues/14
+if(MINGW)
+  target_compile_options(fluid PRIVATE -Wno-narrowing)
+endif()
+
 # 添加一个GUI可执行程序并链接到FLTK库，用法：
 # add_fltk_executable(<name> <source>...)
 function(add_fltk_executable name)

